wp_api/src/wp_com/endpoint.rs

```diff
@@ -14,16 +14,18 @@ pub mod support_bots_endpoint;
 #[derive(uniffi::Object)]
 pub struct WpComDotOrgApiUrlResolver {
     pub base_url: ParsedUrl,
-    pub site_url: String,
+    pub site_id: String,
 }
 
 #[uniffi::export]
 impl WpComDotOrgApiUrlResolver {
     #[uniffi::constructor]
-    pub fn new(site_url: String) -> Self {
+    pub fn new(base_url: Option<String>, site_id: String) -> Self {
```

Comment:
I think using an `enum` such as the one below would be a nicer solution. The function signature with `Option<String>` doesn't make it clear what happens when `None` is passed as an argument without checking the implementation. Using an `enum` would allow us to communicate the information with type signature rather than documentation.

Having said that, the benefit is quite minimal, so I am fine if you want to go with the current implementation.

```
#[derive(Debug, Default, PartialEq, Eq, uniffi::Enum)]
pub enum WpComBaseUrl {
    #[default]
    Default,
    Custom(String),
}
```

Reply:
Good suggestion. I have implemented it in https://github.com/Automattic/wordpress-rs/pull/752/commits/fb4e0c59a0e35c57c0be9184d3ab77b10b8310fa.